### ISSUE TITLE
Error on https://www.numuki.com/gameframe/avie-my-pretty-avatar

### ISSUE THREAD

**Issue by Nolshae** [ruffle-rs/ruffle#13703]
# Error Info
Error name: Error
Error message: panicked at core/src/avm2/amf.rs:99:34:
Unexpected non-object value in object vector
Error stack:
```
Error: panicked at core/src/avm2/amf.rs:99:34:
Unexpected non-object value in object vector
    at imports.wbg.__wbg_new_d258248ed531ff54 (chrome-extension://donbcfbmhbcapadipfkeojnmajbakjdc/dist/339.js:2479:21)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::hee68611ee8147044 (wasm://wasm/032bf7de:wasm-function[9747]:0x995320)
    at std::panicking::rust_panic_with_hook::h3aa054d35a0817d7 (wasm://wasm/032bf7de:wasm-function[6416]:0x8981dd)
    at core::panicking::panic_fmt::h3e1dd3d08288569e (wasm://wasm/032bf7de:wasm-function[9549]:0x990199)
    at core::option::expect_failed::hea22cf135ad64e98 (wasm://wasm/032bf7de:wasm-function[8957]:0x96671e)
    at ruffle_core::avm2::amf::serialize_value::hc18c21f5556707ae (wasm://wasm/032bf7de:wasm-function[797]:0x2a3df4)
    at ruffle_core::avm2::amf::recursive_serialize::h9b172c5e2c026756 (wasm://wasm/032bf7de:wasm-function[2580]:0x62d8ab)
    at ruffle_core::avm2::globals::flash::net::shared_object::new_lso::h4e35c36b477e648a (wasm://wasm/032bf7de:wasm-function[2770]:0x663377)
    at ruffle_core::avm2::globals::flash::net::shared_object::flush::h6d94e2c4fcacb750 (wasm://wasm/032bf7de:wasm-function[1764]:0x4f61e4)
    at ruffle_core::avm2::function::Executable::exec::h12f8e818565825c7 (wasm://wasm/032bf7de:wasm-function[895]:0x31acc4)
```
AVM2 stack:
```
    at flash.net::SharedObject/flash::net::SharedObject::flush()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/updateSharedObject()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/addElementToQueue()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/startSession()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/init()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/setEnvironment()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/init()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/getInstance()
    at com.spilgames.api.core.tracking.ext.spilet::SETTracker/setup()
    at com.spilgames.api.core.tracking.ext::ATracker/setClass()
    at com.spilgames.api.core.tracking.ext::TrackerLoader/load()
    at com.spilgames.api.core.tracking::TrackController/loadLibraries()
    at com.spilgames.api.core.tracking::TrackController/onLoadConfig()
    at com.spilgames.api.core.tracking::TrackConfig/onLoadConfig()
```

# Player Info
Allows script access: false
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (NVIDIA, NVIDIA GeForce RTX 3060 Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: Other
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_COMPRESSION_BC | TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS | SHADER_UNUSED_VERTEX_OUTPUT)
Available features: Features(CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 1000, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: best
Surface samples: 4
Surface size: Extent3d { width: 700, height: 510, depth_or_array_layers: 1 }

# Page Info
Page URL: https://www.numuki.com/gameframe/avie-my-pretty-avatar
SWF URL: https://files.ngdfiles.com/games/girls/avie/Avie_ngdMOD.swf

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 OPR/102.0.0.0
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-10-24
Channel: nightly
Built: 2023-10-24T00:25:33.390Z
Commit: d5da655f8f1374bd0b2acd99cc94d52a2e21a246
Is extension: true

# Metadata
width: 700
height: 510
frameRate: 12
numFrames: 15
swfVersion: 10
backgroundColor: #FFFFFF
isActionScript3: true
uncompressedLength: 14054351

**Comment by n0samu** [comment]
Same error as #12366

**Comment by n0samu** [comment]
Fixed by #11971, but see #14173 for the next issue with this game
